packages/runtime-dom/src/modules/style.ts

```diff
@@ -22,6 +23,12 @@ export function patchStyle(el: Element, prev: Style, next: Style) {
     const currentDisplay = style.display
     if (isCssString) {
       if (prev !== next) {
+        // #9821
+        let cssVarText = (style as any)[CSS_VAR_TEXT]
+        if (cssVarText) {
+          cssVarText = (next.endsWith(';') ? '' : '; ') + cssVarText
```

Comment:
All mainstream browsers are able to handle extra `;` between properties in inline style, so we can just do `;(next as string) += ';' + cssVarText` and make `cssVarText` a const.

Reply:
I have already made the modifications, thank you for your guidance.